testing/test_assertrewrite.py

```diff
@@ -1580,3 +1583,59 @@ def fake_mkdir(p, *, exc):
     with pytest.raises(OSError) as exc_info:
         try_mkdir(str(p))
     assert exc_info.value.errno == errno.ECHILD
+
+
+class TestPyCacheDir:
+    @pytest.mark.parametrize(
+        "prefix, source, expected",
+        [
+            ("c:/tmp/pycs", "d:/projects/src/foo.py", "c:/tmp/pycs/projects/src"),
```

Comment:
@asottile here are the tests for Windows, which I obtained by testing how `sys.pycache_prefix` worked on my system.